rotkehlchen/chain/evm/decoding/balancer/v2/decoder.py

```diff
@@ -105,7 +110,16 @@ def _decode_join_or_exit(self, context: DecoderContext) -> DecodingOutput:
                 event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
                 event.counterparty = CPT_BALANCER_V2
                 event.notes = f'Deposit {event.balance.amount} {token.symbol} to a Balancer v2 pool'  # noqa: E501
-
+                send_events.append(event)
+
+        # in _check_deposits_withdrawals we expect the receive to be the last event.
+        # This happens almost always but there are some cases like rETH on arb where it doesn't.
+        # This reshuffle ensures that the receive event is always the last one before grouping
+        # them in _check_deposits_withdrawals.
+        maybe_reshuffle_events(
+            ordered_events=send_events + receive_events,
+            events_list=context.decoded_events,
+        )
```

Comment:
i was thinking of refactoring the `self._check_deposits_withdrawals()` function but this is simpler.

Reply:
That functions calculates the number of events per balancer pool in the case of multiple deposits. SInce the logic is shared with v1 and I'm not sure if this happens in v1 I believe is just easier to have this split